### PR TITLE
Introduce support for NuttX RTOS

### PR DESCRIPTION
**Pull request by no1wudi**
Initial support for NuttX RTOS, 
the patch for NuttX in my local dev branch is tested and coming soon.

### CODE REVIEW

**Comment by xwang98** [comment]
@no1wudi Excellent work! Could you be the maintainer for the NuttX platform if there is any issues against it?

**Comment by no1wudi** [comment]
> @no1wudi Excellent work! Could you be the maintainer for the NuttX platform if there is any issues against it?

Great pleasure

**Comment by xwang98** [comment]
@no1wudi we will leave the te

> > @no1wudi Excellent work! Could you be the maintainer for the NuttX platform if there is any issues against it?
> 
> Great pleasure

Great! We will leave the validation and maintenance of NuttX platform to you. :-) Please send a notice when you think this patch is ready to merge.

File: product-mini/platforms/nuttx/test_wasm.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 XiaoMi Corporation.  All rights reserved.
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/test_wasm.h:2]: This file is copied from WAMR project, could you please change the copyright?

File: product-mini/platforms/nuttx/wamr.mk [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+else
+CFLAGS += -DWASM_ENABLE_LIBC_BUILTIN=0
+endif
+
+CFLAGS += -DWASM_ENABLE_MULTI_MODULE=0
+CFLAGS += -DWASM_ENABLE_THREAD_MGR=0
+CFLAGS += -Wno-strict-prototypes
+
+CFLAGS += -I${CORE_ROOT} \
+		  -I${IWASM_ROOT}/include \
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/wamr.mk:92]: Had better modify the code indent

**Review by wenyongh — COMMENTED** [review]
Could you please also update README.md to add NuttX to supported platform list, and update doc/build_wamr.md to add how to build and run in the platform?

File: product-mini/platforms/nuttx/test_wasm.h (hunk 1 above)
Comment (no1wudi) [reply product-mini/platforms/nuttx/test_wasm.h]: Done

**Comment by no1wudi** [comment]
> Could you please also update README.md to add NuttX to supported platform list, and update doc/build_wamr.md to add how to build and run in the platform?


Document added.
For NuttX, WAMR works as a builtin application, the sources and building would be managed by NuttX's building system.

**Pull request merged**